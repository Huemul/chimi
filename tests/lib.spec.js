const { stripIndent } = require('common-tags')
const { injectDependencies } = require('../lib/lib')

describe('lib', () => {
  describe('injectDependencies', () => {
    const code = stripIndent`
      const answer = 42

      console.log(answer)
    `

    it('should only add the source maps code when no dependencies are passed', () => {
      const result = injectDependencies({}, code)

      expect(result).toMatchSnapshot()
    })

    it('should not change snippet indentation', () => {
      const withIndentation = stripIndent`
        const foo = {
          bar: "bar"
        };

        Promise.resolve(foo)
          .then(({ bar }) => bar)
          .then(bar => {
            console.log(bar);
          });
      `
      let result = injectDependencies({}, withIndentation)

      expect(result).toMatchSnapshot()

      const dependencies = {
        lodash: '_',
        './lib': 'lib',
      }
      result = injectDependencies(dependencies, withIndentation)

      expect(result).toMatchSnapshot()
    })

    it('should prepend requires with variable declarations to the input', () => {
      let dependencies = {
        lodash: '_',
      }
      let result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()

      dependencies = {
        lodash: '_',
        trae: 'trae',
      }
      result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()
    })

    it('should work for local dependencies', () => {
      const dependencies = {
        './lib': 'lib',
        '../foo/bar': 'bar',
      }
      const result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()
    })

    it('should not add a variable declaration when dependencies map values are falsy', () => {
      const dependencies = {
        './for-the-side-effects': false,
        'es6-promise': false,
      }
      const result = injectDependencies(dependencies, code)

      expect(result).toMatchSnapshot()
    })
  })
})
