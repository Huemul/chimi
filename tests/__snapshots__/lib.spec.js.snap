// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib injectDependencies should add globals 1`] = `
"require('source-map-support').install()
// snippet dependencies

let theAnswer = 42;

const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7OztBQUFBO0FBQ0E7QUFDQSIsImZpbGUiOiIvZGVjb3JhdGVkLXNuaXBwZXQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJjb25zdCBhbnN3ZXIgPSA0MlxuXG5jb25zb2xlLmxvZyhhbnN3ZXIpIl19"
`;

exports[`lib injectDependencies should add globals and dependencies 1`] = `
"require('source-map-support').install()
// snippet dependencies
const _ = require('lodash');
let theAnswer = 42;

const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7OztBQUFBO0FBQ0E7QUFDQSIsImZpbGUiOiIvZGVjb3JhdGVkLXNuaXBwZXQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJjb25zdCBhbnN3ZXIgPSA0MlxuXG5jb25zb2xlLmxvZyhhbnN3ZXIpIl19"
`;

exports[`lib injectDependencies should add requires with no declaration when the name is missing 1`] = `
"require('source-map-support').install()
// snippet dependencies
require('./for-the-side-effects');
require('es6-promise');


const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7QUFBQTtBQUNBO0FBQ0EiLCJmaWxlIjoiL2RlY29yYXRlZC1zbmlwcGV0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiY29uc3QgYW5zd2VyID0gNDJcblxuY29uc29sZS5sb2coYW5zd2VyKSJdfQ=="
`;

exports[`lib injectDependencies should add source maps but no dependencies 1`] = `
"require('source-map-support').install()
// snippet dependencies



const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7OztBQUFBO0FBQ0E7QUFDQSIsImZpbGUiOiIvZGVjb3JhdGVkLXNuaXBwZXQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJjb25zdCBhbnN3ZXIgPSA0MlxuXG5jb25zb2xlLmxvZyhhbnN3ZXIpIl19"
`;

exports[`lib injectDependencies should interpolate functions, arrays, objets and strings properly 1`] = `
"require('source-map-support').install()
// snippet dependencies

let add = (a, b) => a + b;
let message = 'hello world';
let foo = { foo: 'bar' };
let fruits = ['orange', 'apple'];

const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7OztBQUFBO0FBQ0E7QUFDQSIsImZpbGUiOiIvZGVjb3JhdGVkLXNuaXBwZXQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJjb25zdCBhbnN3ZXIgPSA0MlxuXG5jb25zb2xlLmxvZyhhbnN3ZXIpIl19"
`;

exports[`lib injectDependencies should not change snippet indentation 1`] = `
"require('source-map-support').install()
// snippet dependencies



const foo = {
  bar: \\"bar\\"
};

Promise.resolve(foo)
  .then(({ bar }) => bar)
  .then(bar => {
    console.log(bar);
  });
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7OztBQUFBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQSIsImZpbGUiOiIvZGVjb3JhdGVkLXNuaXBwZXQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJjb25zdCBmb28gPSB7XG4gIGJhcjogXCJiYXJcIlxufTtcblxuUHJvbWlzZS5yZXNvbHZlKGZvbylcbiAgLnRoZW4oKHsgYmFyIH0pID0+IGJhcilcbiAgLnRoZW4oYmFyID0+IHtcbiAgICBjb25zb2xlLmxvZyhiYXIpO1xuICB9KTsiXX0="
`;

exports[`lib injectDependencies should not change snippet indentation 2`] = `
"require('source-map-support').install()
// snippet dependencies
const _ = require('lodash');
const lib = require('./lib');


const foo = {
  bar: \\"bar\\"
};

Promise.resolve(foo)
  .then(({ bar }) => bar)
  .then(bar => {
    console.log(bar);
  });
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7QUFBQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0EiLCJmaWxlIjoiL2RlY29yYXRlZC1zbmlwcGV0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiY29uc3QgZm9vID0ge1xuICBiYXI6IFwiYmFyXCJcbn07XG5cblByb21pc2UucmVzb2x2ZShmb28pXG4gIC50aGVuKCh7IGJhciB9KSA9PiBiYXIpXG4gIC50aGVuKGJhciA9PiB7XG4gICAgY29uc29sZS5sb2coYmFyKTtcbiAgfSk7Il19"
`;

exports[`lib injectDependencies should prepend requires with variable declarations to the snippet code 1`] = `
"require('source-map-support').install()
// snippet dependencies
const _ = require('lodash');


const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7OztBQUFBO0FBQ0E7QUFDQSIsImZpbGUiOiIvZGVjb3JhdGVkLXNuaXBwZXQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJjb25zdCBhbnN3ZXIgPSA0MlxuXG5jb25zb2xlLmxvZyhhbnN3ZXIpIl19"
`;

exports[`lib injectDependencies should prepend requires with variable declarations to the snippet code 2`] = `
"require('source-map-support').install()
// snippet dependencies
const _ = require('lodash');
const trae = require('trae');


const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7QUFBQTtBQUNBO0FBQ0EiLCJmaWxlIjoiL2RlY29yYXRlZC1zbmlwcGV0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiY29uc3QgYW5zd2VyID0gNDJcblxuY29uc29sZS5sb2coYW5zd2VyKSJdfQ=="
`;

exports[`lib injectDependencies should work for local dependencies 1`] = `
"require('source-map-support').install()
// snippet dependencies
const lib = require('./lib');
const bar = require('../foo/bar');


const answer = 42

console.log(answer)
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlJFQURNRS5tZCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7QUFBQTtBQUNBO0FBQ0EiLCJmaWxlIjoiL2RlY29yYXRlZC1zbmlwcGV0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiY29uc3QgYW5zd2VyID0gNDJcblxuY29uc29sZS5sb2coYW5zd2VyKSJdfQ=="
`;

exports[`lib listDependencies should list dependencies in different types 1`] = `
"const trae = require('trae');
require('es6-promise');
const _ = require('lodash');
let config = require('./config');"
`;
